language/src/main.rs

```diff
@@ -318,5 +348,36 @@ fn main() -> Result<(), ()> {
     match RunCompiler::new(&args, &mut callbacks).run() {
         Ok(_) => Ok(()),
         Err(_) => Err(()),
-    }
+    }?;
+    
+    match output_file_clone.clone() {
+        Some (file) => {
+            match &output_template_file {
+                None => {
+                    if copy_template {
+                        std::fs::copy(file.clone(), file.clone() + "_template").expect("Copy failed");
+                    };
```

Comment:
I would be in favor of putting that into a `Makefile` rather than inside the tool itself; this is more of a build system concern that something to do with the compilation.

Reply:
Since the tool runs for Cargo and this is a feature everyone using the system

Reply:
Since the tool is build with `Cargo` I thought I would use that as a build system, even though I have more experience with `make`, but I can try and move it into a `Makefile`.